src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -328,25 +327,11 @@ func (cfg Config) cloudPollSleepTime() time.Duration {
 	return time.Duration((CloudConfigPollInterval.Nanoseconds() / 2) + rand.Int63n(CloudConfigPollInterval.Nanoseconds()))
 }
 
-func (cfg Config) fetchCloudConfig() (bytes []byte, err error) {
+func (cfg Config) fetchCloudConfig(withClient func(func(*http.Client))) (bytes []byte, err error) {
 	log.Debugf("Fetching cloud config...")
-
-	if cfg.IsDownstream() {
-		// Clients must always proxy the request
-		dialer := cfg.Client.HighestQOSFrontedDialer()
-		// We use direct domain fronting for accessing new configs.
-		client := dialer.NewDirectDomainFronter()
-		bytes, err = cfg.fetchCloudConfigForClient(client)
-	} else {
-		var client *http.Client
-		// In the server side, we use a direct (non-proxied) client.
-		if client, err = util.HTTPClient(cfg.CloudConfigCA, ""); err == nil {
-			// We need to ask for HTTPS explicitly for these.
-			bytes, err = cfg.fetchCloudConfigForClient(client)
-		} else {
-			err = fmt.Errorf("Unable to initialize HTTP client: %s", err)
-		}
-	}
+	withClient(func(c *http.Client) {
```

Comment:
While `withClient` is always the same function for the lifetime of the Lantern client, `c` is always the latest fronter.